lib/chef/data_collector.rb

```diff
@@ -50,8 +50,8 @@ def self.register_reporter?
                         "#{Chef::Config[:data_collector][:mode].inspect} modes, disabling it")
         return false
       end
-      unless data_collector_url_configured?
-        Chef::Log.debug("data collector URL is not configured, disabling data collector")
+      unless data_collector_url_configured? || data_collector_output_locations_configured?
```

Comment:
it seems odd, throughout this, to special case data_collector_url. Instead, might it be better to just add that url to the list of output locations?

Reply:
That is entirely possible and something I considered, however there is a bunch of behaviour for that config setting specific to being used with automate, like the initial post & disable behaviour. I erred on the side of leaving that part untouched and augmenting it with output_locations, but I'm entirely happy to merge that all in to one if we feel it'd be better.